### PR TITLE
Namadillo: reviewing all notifications

### LINKED ISSUE #976: Namadillo: repeated errors only show one error notification
When you do something in Namadillo that raises an error notification, if you do the same thing again there is no new error notification. The second error should raise a notification, and the first notification should either be kept or replaced.

One example is when there is an error in a staking transaction.

I'm guessing this behaviour is because the notifications do not have unique IDs. This is probably a problem in a lot of different places and fixing will require checking all the error notification IDs.

### LINKED ISSUE #977: Namadillo: Trying to stake to multiple validators where one amount is 0 throws an error
If you try to stake to multiple validators and one of the amounts is 0, an error is thrown. We can probably safely filter out the validators where the amount is 0. This edge case could come up if a user changes their mind about how much to stake and sets it from a valid number to 0.

### LINKED ISSUE #978: Namadillo: Staking/unstaking multiple validators success notification only shows one validator
When staking or unstaking from multiple validators, the success notification only contains information about one of the validators.

### PR DESCRIPTION
**Pull request by pedrorezende**
Reviewing the way notifications are fired on Staking and Governance flows:
- Adding more info to notifications
- Parsing batch transactions
- Restarts timeout when user hovers the toast

Fixes #976
Fixes #977 
Fixes #978

### CODE REVIEW

**Review by euharrison — APPROVED** [review]
Niiiice! LGTM!

Really small tweaks:

- On error, the tx id is wider than the box
Screenshot 2024-08-13 at 12 13 00

- Can we remove this warning when expanding the details?
Screenshot 2024-08-13 at 12 12 08

**Review by euharrison — APPROVED** [review]

**Review by emccorson — APPROVED** [review]
Looks good to me! I especially like the timeouts part :sunglasses:

File: apps/namadillo/src/App/Staking/IncrementBonding.tsx [hunk 1]
```diff
@@ -127,26 +131,23 @@ const IncrementBonding = (): JSX.Element => {
...
+      dispatchPendingNotification(bondTransactionData);
       onCloseModal();
     }
   }, [isSuccess]);
 
   useEffect(() => {
     if (isError) {
       dispatchNotification({
-        id: "staking-error",
+        id: createNotificationId(),
```
[the remark this answers is not in the document]
Comment (pedrorezende) [reply apps/namadillo/src/App/Staking/IncrementBonding.tsx:143]: Correct if I'm wrong, but I think both notifications will occur in different moments. The one located in the component will handle eventual failures on communicating with the blockchain / user rejecting signature, etc. The one fired by the event handles anything related to the transaction processing in the chain, like out of gas, for example

File: apps/namadillo/src/App/Common/Toast.tsx [hunk 2]
```diff
@@ -76,10 +97,29 @@ const Toast = ({ notification, onClose }: ToastProps): JSX.Element => {
...
+          "items-start relative break-words overflow-hidden",
+          "max-w-full leading-normal"
+        )}
+      >
+        <strong className="block text-sm">{notification.title}</strong>
         <div className="leading-tight text-xs">{notification.description}</div>
-      </div>
+        {notification.details && !viewDetails && (
+          <button
+            className="text-xs text-white leading-0 underline"
```
[the remark this answers is not in the document]
Comment (pedrorezende) [reply apps/namadillo/src/App/Common/Toast.tsx:111]: Ops. It should be `leading-none`, however, this is just something temporary I was using to test one thing. Forgot to remove. Doing it now, thanks!

File: apps/namadillo/src/App/Staking/IncrementBonding.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (pedrorezende) [reply apps/namadillo/src/App/Staking/IncrementBonding.tsx:143]: @emccorson I'm going to proceed and merge this one, but we could keep this discussion in a separate issue or on Slack

**Pull request merged**